scylla/src/routing.rs

```diff
@@ -6,7 +6,26 @@ use thiserror::Error;
 
 #[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Copy, Debug)]
 pub struct Token {
-    pub value: i64,
+    value: i64,
+}
+
+impl Token {
+    #[inline]
+    pub fn new(value: i64) -> Self {
+        Self {
+            value: if value == i64::MIN { i64::MAX } else { value },
+        }
+    }
+
+    #[inline]
+    pub fn new_denormalized(value: i64) -> Self {
+        Self { value }
+    }
```

Comment:
I exposed all methods because before the change the field itself was public. We could restrict this method (maybe `new()` too?), but I'd like to know what users need `Token` for. Is there any plausible scenario where they need to create a new token?

Reply:
I can't think of any such scenario. WDYT @piodul?

Reply:
Looking at the code below, it seems there is one case where you call new_denormalize() but only for the MIN value itself. So I think instead of new_denormalized(i64) you should have just a new_invalid() which returns an invalid token - it it64:min.

Reply:
That's right, we either want to create a valid token (everywhere else) or specifically an invalid token (`i64::MIN`).

Reply:
I'm not sure anything comes to my mind as far as real use cases for users go for `new_denormalized`. For our internal use at least we will have to represent token ranges somehow, specifically their left ends. I don't know whether this concern for users.

Reply:
I'm worried that a year from now, nobody will remember what a "denormalized" token means. So you'll either need to write a comment explaining what this constructor means and when to use it - or just not have this constructor and have a separate new_invalid() just for the special case of min_token which is easy to understand what it means. If it's enough...
BTW, If I understand correctly, an "invalid token" (min_int) will never be part of any token range.

Reply:
> I'm not sure anything comes to my mind as far as real use cases for users go for `new_denormalized`. For our internal use at least we will have to represent token ranges somehow, specifically their left ends. I don't know whether this concern for users.

What I meant is: are there are scenarios where user needs to create a new token from integer? I'm not talking just about invalid tokens, but any tokens at all. @piodul 
I'm asking because I'm wondering if we can make all constructors private.

Reply:
I don't know, but I'd err on the more cautious side and allow creating tokens from integers.

Do we have any APIs in the driver that accept `Token`?

Reply:
> I don't know, but I'd err on the more cautious side and allow creating tokens from integers.
> 
> Do we have any APIs in the driver that accept `Token`?

There are some:

https://docs.rs/scylla/latest/scylla/routing/struct.Sharder.html#method.shard_of
https://docs.rs/scylla/latest/scylla/transport/locator/struct.TokenRing.html#method.get_elem_for_token
https://docs.rs/scylla/latest/scylla/transport/locator/struct.TokenRing.html#method.ring_range
https://docs.rs/scylla/latest/scylla/transport/locator/struct.TokenRing.html#method.ring_range_full
https://docs.rs/scylla/latest/scylla/transport/locator/struct.ReplicaLocator.html#method.replicas_for_token
https://docs.rs/scylla/latest/scylla/transport/struct.ClusterData.html#method.get_token_endpoints


I found them by search `Token` on docs.rs and going to "In Parameters" tab - judging by the results I got I think those should be exhaustive.

Reply:
So the constructor needs to remain public I think. Not sure about INVALID constant (previously new_unchecked). I think we can make it private.